modules/table/src/main/java/org/apache/ignite/internal/table/distributed/index/IndexMetaStorage.java

```diff
@@ -154,78 +169,86 @@ Collection<IndexMeta> indexMetasSnapshot() {
         return List.copyOf(indexMetaByIndexId.values());
     }
 
-    private void onCatalogIndexCreateEvent(CreateIndexEventParameters parameters) {
+    private CompletableFuture<Boolean> onCatalogIndexCreateEvent(CreateIndexEventParameters parameters) {
         CatalogIndexDescriptor catalogIndexDescriptor = parameters.indexDescriptor();
 
         Catalog catalog = catalog(parameters.catalogVersion());
 
-        updateAndSaveIndexMetaToVault(catalogIndexDescriptor.id(), indexMeta -> {
-            assert indexMeta == null : catalogIndexDescriptor.id();
+        return updateAndSaveIndexMetaToMetastore(catalogIndexDescriptor.id(), indexMeta -> {
+            assert indexMeta == null : "indexId=" + catalogIndexDescriptor.id() + "catalogVersion=" + catalog.version();
 
-            return new IndexMeta(catalogIndexDescriptor, catalog);
-        });
+            return IndexMeta.of(catalogIndexDescriptor, catalog);
+        }).thenApply(unused -> false);
```

Comment:
Why all futures always complete with false? Can't wrap my head around it

Reply:
Please read javadoc of `org.apache.ignite.internal.event.EventProducer` and `org.apache.ignite.internal.event.EventListener` for understanding.